source/ChiPSeqReader.py

```diff
@@ -362,6 +362,32 @@ def duplicate_region(self, interval):
         self.chr_data[interval.chr].sort_values(by=["start"], inplace=True)
         assert len(self.chr_data[interval.chr]) - debug == old_length
 
+    def duplicate_region_v2(self, interval):                     # Modifies data according to the relationship
+        st, en = self.get_interval(interval, return_ids=True)    # between the "start" and "end" of each peak and
+        #print(st, en)                                           # the "start" and "end" of the duplication interval
+        old_length = len(self.chr_data[interval.chr])
+        drop_indices = []
+        dup_indices = []
+        for i in range(st, en + 1):
+            if (self.chr_data[interval.chr].iloc[i, self.chr_data[interval.chr].columns.get_loc("start")] < interval.start) and \
+                    (self.chr_data[interval.chr].iloc[i, self.chr_data[interval.chr].columns.get_loc("end")] > interval.end):
+                drop_indices.append(i)                # Duplication interval is within peak interval. Gets indices of destroyed peaks
+            elif (self.chr_data[interval.chr].iloc[i, self.chr_data[interval.chr].columns.get_loc("start")] > interval.start) and \
+                    (self.chr_data[interval.chr].iloc[i, self.chr_data[interval.chr].columns.get_loc("end")] < interval.end):
+                dup_indices.append(i)                 # Peak interval is within duplication interval. Gets indices of duplicated peaks
+        debug = len(dup_indices) - len(drop_indices)
+        dup_data = self.chr_data[interval.chr].loc[self.chr_data[interval.chr].index[dup_indices]]  # Duplicated peaks as df
+        dup_data["start"] += interval.len
+        dup_data["end"] += interval.len
+        dup_data["mids"] += interval.len
+        self.chr_data[interval.chr].iloc[en:, self.chr_data[interval.chr].columns.get_loc("start")] += interval.len
+        self.chr_data[interval.chr].iloc[en:, self.chr_data[interval.chr].columns.get_loc("end")] += interval.len
+        self.chr_data[interval.chr].iloc[en:, self.chr_data[interval.chr].columns.get_loc("mids")] += interval.len
+        self.chr_data[interval.chr].drop(self.chr_data[interval.chr].index[drop_indices], inplace=True)  # Deletes destroyed peaks
```

Comment:
I think this does not handle item "en" correctly. It will first add it to dup_data df, and increase it's coordinate at lines 380-382. Next, it will change the coordinate in original df at lines 383-385. Thus, there will be not peak at position en after executing this code.

Reply:
I used an old function "get_interval" and it seems to me that item "en", obtained during its application, allows us to get the correct results. 
Writing the code, I was not aware of the existence of the new function.  In the nearest future I plan to change the application of the old function "get_interval" to the application of the new one.